### PR TITLE
ml - Filter images by tag

### LINKED ISSUE #7: Images by Tag
#### As a user I want to view all images associated with a tag.

__Story__:

Now that we have added the ability for images to be associated with tags, let
us provide the ability for users to filter images by a given tag. Nana, who
loves fast planes, will be delighted to see only images labeled with SR-71.

__Acceptance criteria__:
- [ ] When I click on a tag, I see a filtered list of only the images that have
  the tag I clicked on.
- [ ] The tag name, not its ID, is used to perform the filtering.

__Discussion topics__:
- [ ] Where in the code should we implement this functionality?
- [ ] What happens if a user tries to manually filter by a nonexistent tag?

__Dependencies__:
- Image Tags

### PR DESCRIPTION
**Pull request by mluders**
Closes #7

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [x] I have replaced `X` above with the associated issue number.
- [x] I have changed the issue's label to "pull request".
- [x] I have assigned the designated reviewer to this PR.
- [x] I have set the appropriate label on this PR.
- [x] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

File: app/controllers/images_controller.rb [hunk 1]
```diff
@@ -1,6 +1,15 @@
 class ImagesController < ApplicationController
   def index
-    @images = Image.order(created_at: :desc)
+    # get the query string
+    # filter
+    tag = params['tag']
+    @images = begin
```
Comment (HaominZhang) [inline app/controllers/images_controller.rb]: Do we need this `begin` here?

Comment (HaominZhang) [inline app/controllers/images_controller.rb]: Do we need this `begin` here?

Comment (mluders) [reply app/controllers/images_controller.rb]: I actually learned this technique from Steven Boyd. He suggested I use a `begin` in order to initialize `@images` with a single `=` operator.

My original code looked like this:
```
if tag.present?
  @images = Image.tagged_with(tag).order(created_at: :desc)
else
  @images = Image.order(created_at: :desc)
end
```

What do you think about this?

Comment (HaominZhang) [reply app/controllers/images_controller.rb]: I think we could definitely use `begin` here, but to me it is a little overkill here. If I remember correctly, we can directly write this:
```
@images = if tag.present?
                        Image.tagged_with(tag).order(created_at: :desc)
                   else
                       Image.order(created_at: :desc)
                   end
```
But maybe there is something else beneficial that I am not aware of

Comment (mluders) [reply app/controllers/images_controller.rb]: Oh right! I'm overcomplicating it.

**Pull request merged**